chapters/Syntax.md

```diff
@@ -1,36 +1,225 @@
+1. If the value of the nibble is hexadecimal `F` (decimal `15`), advance to the
+   next nibble (the lower nibble of the current byte or the higher nibble of the
+   next byte) and return to the previous step.
+
+More simply:
+
+```text
+version = 1 + sum of all nibbles up to and including first non-hexadecimal-F
+```
+
+What appears beyond the version is version-specific. Implementations MUST NOT
+attempt to interpret **did:btc1** identifiers with an unknown `version`.
+
+For illustration purposes only, assume that the nibble following the version
+always represents the network and that the allowable values for the network
+don't change. The interpretation of the`version` and `network` would be as
+follows (spaces between bytes added for readability):
+
+| Decoded bytes     | Version | Network         |
+|-------------------|--------:|-----------------|
+| *00* 03 c7 ...    |       1 | bitcoin         |
+| *23* 03 c7 ...    |       3 | testnet v3      |
+| *C9* 03 c7 ...    |      13 | custom signet 2 |
+| *F2 90* 03 c7 ... |      18 | custom signet 2 |
+| *FF 72* 03 c7 ... |      38 | regtest         |
+
+### Encoding a **did:btc1** Identifier
+
+Given:
+
+* `idType` - one of:
+  * "key"
+  * "hash"
+* `version` - number
+* `network` - one of:
+  * "bitcoin"
+  * "signet"
+  * "regtest"
+  * "testnet3"
+  * "testnet4"
+  * number
+* `idBytes` - byte array, one of:
+  * a compressed secp256k1 public key if `idType` is "key"
+  * a hash of an initiating DID document if `idType` is "hash"
+
+Encode the **did:btc1** identifier as follows:
+
+1. If `idType` is not a valid value per above, raise `InvalidDID` error.
+1. If `version` is not `1`, raise `InvalidDID` error.
+1. If `network` is not a valid value per above, raise `InvalidDID` error.
+1. if `network` is a number and is outside the range of 1-8, raise `InvalidDID` error.
+1. If `idType` is "key" and `idBytes` is not a valid compressed secp256k1
+   public key, raise `InvalidDID` error.
+1. Map `idType` to `hrp` from the following:
+   1. "key" - "k"
+   1. "hash" - "x"
+1. Create an empty `nibbles` numeric array.
+1. Set `fCount` equal to `(version - 1) / 15`, rounded down.
+1. Append hexadecimal `F` (decimal `15`) to `nibbles` `fCount` times.
+1. Append `(version - 1) mod 15` to `nibbles`.
+1. If `network` is a string, append the numeric value from the following map to
+   `nibbles`:
+    1. "bitcoin" - `0`
+    1. "signet" - `1`
+    1. "regtest" - `2`
+    1. "testnet3" - `3`
+    1. "testnet4" - `4`
+1. If `network` is a number, append `network + 7` to `nibbles`.
+1. If the number of entries in `nibbles` is odd, append `0`.
+1. Create a `decodedBytes` byte array from `nibbles`, where `index` is from `0`
+   to `nibbles.length / 2 - 1` and `encodingBytes[index] = (nibbles[2 * index]
+   << 8) | nibbles[2 * index + 1]`.
+1. Append `idBytes` to `encodingBytes`.
+1. Set `id` to "did:btc1:".
+1. Pass `hrp` and `decodedBytes` to the Bech32m encoding algorithm, retrieving
```

Comment:
There is currently a section in the appendix for the bech32 encode and decode algorithms.

We should either reference them or remove them.